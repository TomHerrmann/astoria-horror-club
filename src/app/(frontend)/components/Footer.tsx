import CopyrightText from './CopyrightText'
import SocialLinks from './SocialLinks'

/**
 *
 * @returns Global Footer coponent for AHC app
 */
export default function Footer() {
  return (
    <footer className="flex flex-col justify-between items-center">
      <SocialLinks />
      <CopyrightText />
    </footer>
  )
}
