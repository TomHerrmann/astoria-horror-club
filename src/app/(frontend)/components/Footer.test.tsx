import { describe, it, expect, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import Footer from './Footer'

// Mock child components
vi.mock('./SocialLinks', () => ({
  default: () => <div data-testid="social-links-mock" />,
}))

vi.mock('./CopyrightText', () => ({
  default: () => <div data-testid="copyright-text-mock" />,
}))

describe('Footer', () => {
  it('matches snapshot', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })

  it('renders the footer with correct structure', () => {
    render(<Footer />)

    const footer = screen.getByRole('contentinfo')
    expect(footer).toBeDefined()
  })

  it('renders the SocialLinks component', () => {
    render(<Footer />)
    expect(screen.getByTestId('social-links-mock')).toBeDefined()
  })

  it('renders the CopyrightText component', () => {
    render(<Footer />)
    expect(screen.getByTestId('copyright-text-mock')).toBeDefined()
  })
})
