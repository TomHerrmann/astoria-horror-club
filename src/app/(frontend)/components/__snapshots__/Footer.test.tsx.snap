// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > matches snapshot 1`] = `
<div>
  <footer
    class="flex flex-col justify-between items-center"
  >
    <div
      data-testid="social-links-mock"
    />
    <div
      data-testid="copyright-text-mock"
    />
  </footer>
</div>
`;
